ta/arch/arm/ta.ld.S

```diff
@@ -8,14 +8,20 @@ OUTPUT_ARCH(aarch64)
 #endif
 
 PHDRS {
-	rodata PT_LOAD;
+	/*
+	 * Exec and rodata headers are hard coded to RX and RO
+	 * respectively. This is needed because the binary is relocatable
+	 * and the linker would automatically make any header writeable
+	 * that need to be updated during relocation.
+	 */
+	exec PT_LOAD FLAGS (5);		/* RX */
+	rodata PT_LOAD FLAGS (4);	/* RO */
 	rwdata PT_LOAD;
-	rodata2 PT_LOAD;
 	dyn PT_DYNAMIC;
 }
 
 SECTIONS {
-	.ta_head : {*(.ta_head)} :rodata
+	.ta_head : {*(.ta_head)} :exec
```

Comment:
I see, thanks for clarifying. I still find :exec here to be a bit disturbing, it's like we are explicitly asking for the header to be executable, but it's not a big deal.